File: src/test/java/com/raychenon/leetcode/array/SubarraySumsDivisibleByKTest.java
package com.raychenon.leetcode.array;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

public class SubarraySumsDivisibleByKTest {

    private static SubarraySumsDivisibleByK subarraySumsDivisibleByK;

    @BeforeClass
    public static void setUp() {
        subarraySumsDivisibleByK = new SubarraySumsDivisibleByK();
    }


    @Test
    public void assertTest1() {
        var array = new int[]{4, 5, 0, -2, -3, 1};
        assertEquals(7, subarraySumsDivisibleByK.subarraysDivByK(array, 5));
    }

    @Test
    public void assertWithOneElement() {
        var array = new int[]{5};
        assertEquals(0, subarraySumsDivisibleByK.subarraysDivByK(array, 9));
    }

    @Test
    public void assertWithOneElementRightDivider() {
        var array = new int[]{5};
        assertEquals(1, subarraySumsDivisibleByK.subarraysDivByK(array, 5));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/array/SubarraySumsDivisibleByKTest.kt
package com.raychenon.kotlin.leetcode.array

import junit.framework.TestCase
import org.junit.Test

class SubarraySumsDivisibleByKTest {


    @Test
    fun assertTest1() {
        val array = intArrayOf(4, 5, 0, -2, -3, 1)
        TestCase.assertEquals(7, SubarraySumsDivisibleByK.subarraysDivByK(array, 5))
    }

    @Test
    fun assertWithOneElement() {
        val array = intArrayOf(5)
        TestCase.assertEquals(0, SubarraySumsDivisibleByK.subarraysDivByK(array, 9))
    }

    @Test
    fun assertWithOneElementRightDivider() {
        val array = intArrayOf(5)
        TestCase.assertEquals(1, SubarraySumsDivisibleByK.subarraysDivByK(array, 5))
    }

}